src/main/java/graphql/normalized/ExecutableNormalizedOperationToAstCompiler.java

```diff
@@ -185,12 +222,37 @@ private static SelectionSet selectionSet(List<Field> fields) {
     }
 
     @NotNull
-    private static Value<?> argValue(ExecutableNormalizedField executableNormalizedField, String argName, NormalizedInputValue normalizedInputValue, VariableAccumulator variableAccumulator) {
+    private static Value<?> argValue(ExecutableNormalizedField executableNormalizedField,
+                                     String argName,
+                                     NormalizedInputValue normalizedInputValue,
+                                     VariableAccumulator variableAccumulator) {
         if (variableAccumulator.shouldMakeVariable(executableNormalizedField, argName, normalizedInputValue)) {
             VariableValueWithDefinition variableWithDefinition = variableAccumulator.accumulateVariable(normalizedInputValue);
             return variableWithDefinition.getVariableReference();
         } else {
             return argValue(executableNormalizedField, argName, normalizedInputValue.getValue(), variableAccumulator);
         }
     }
+
+    @Nullable
+    private static GraphQLFieldDefinition getFieldDefinition(GraphQLSchema schema,
+                                                             String parentType,
+                                                             ExecutableNormalizedField nf) {
+        return schema.getFieldDefinition(coordinates(parentType, nf.getName()));
+    }
+
+    @Nullable
+    private static GraphQLObjectType getOperationType(@NotNull GraphQLSchema schema,
```

Comment:
I am sure we have the same code 5 times already somewhere else :-)

Reply:
Fixed